src/python/pants/backend/python/lint/black/subsystem_test.py

```diff
@@ -29,12 +29,12 @@ def test_setup_lockfile_interpreter_constraints() -> None:
 
     global_constraint = "==3.9.*"
     rule_runner.set_options(
-        ["--black-lockfile=lockfile.txt"],
+        ["--black-lockfile=lockfile.txt", "--no-python-infer-imports"],
```

Comment:
Setting this fixes the tests to work on my M1 which doesn't have <Py38. Now we don't try to parse imports, which were empty anyways.

Reply:
Not sure if relevant, but: in some cases, setting `env_inherit={'PATH'}` will allow pyenv, etc to be picked up (which is why you see that scattered all over).

Reply:
The issue specifically is when we set things like `==3.7` rather than loose ranges like `>=3.7`. But either way, this is a good change: the tests work on more machines, don't need env vars, and are faster b/c no subprocesses.